### PR TITLE
fix: leaderboard functions

### PR DESCRIPTION
**Pull request by Faf4a**
## Type
- [x] Bug Fix
- [x] Functions: Fixes leaderboard functions

Referenced Issue: #565 

## Description
Fixed leaderboard functions, and removes a check if users dont exist.

### CODE REVIEW

**Comment by Leref** [comment]
I’m guessing if we don’t provide hideZero value user, it’ll just say the replace message placeholder.
Otherwise everything looks good.

**Comment by Leref** [comment]
Also, make sure we update all leaderboard functions with the same placeholder to make it consistent with all functions.

File: src/functions/database/globalUserLeaderBoard.js [hunk 1]
```diff
@@ -1,54 +1,59 @@
...
-            "{tag}": user.tag,
-            "{id}": user.id,
-            "{mention}": user,
-        }
+            "{username}": user?.username || "Unknown User",
+            "{displayName}": user?.displayName || "Unknown User",
+            "{tag}": user?.tag || "Unknown User",
+            "{id}": user?.id || "Unknown User",
+            "{mention}": user || "Unknown User"
+        };
```
Comment (USERSATOSHI) [inline src/functions/database/globalUserLeaderBoard.js]: just ignore all non existing user one
as uh other var types can also be mixed in 

if the user is using same var for different var type

File: src/functions/database/guildLeaderBoard.js [hunk 2]
```diff
@@ -1,50 +1,55 @@
...
+
+        if (hideNegativeValue === "true" && parseInt(lbdata.value) < 0) continue;
+        if (hideZeroValue === "true" && parseInt(lbdata.value) == 0) continue;
+
+        const replacer = {
+            "{value}": parseInt(lbdata.value),
+            "{top}": i + 1,
+            "{name}": guild?.name || "Unknown Guild",
+            "{id}": guild.id || "Unknown Guild"
+        };
```
Comment (USERSATOSHI) [inline src/functions/database/guildLeaderBoard.js]: same here
ignore one that cant be found

File: src/functions/database/userLeaderBoard.js [hunk 3]
```diff
@@ -1,60 +1,73 @@
...
+    for (const lbdata of db) {
+        const key = lbdata.key.split("_")[1];
+        let member = await d.util.getMember(guild, key);
+
+        if (!member) member = await d.util.getUser(d, key);
+
+        if (hideNegativeValue === "true" && parseInt(lbdata.value) < 0) continue;
+        if (hideZeroValue === "true" && parseInt(lbdata.value) == 0) continue;
+
+        const replacer = {
```
Comment (USERSATOSHI) [inline src/functions/database/userLeaderBoard.js]: we could do it here too but this one can be ignored as only uservar uses userid and guildid so it should be fine

**Review by USERSATOSHI — CHANGES_REQUESTED** [review]

File: src/functions/database/userLeaderBoard.js [hunk 4]
```diff
@@ -35,19 +35,20 @@ module.exports = async (d) => {
         let member = await d.util.getMember(guild, key);
 
         if (!member) member = await d.util.getUser(d, key);
+        if (!member) continue;
```
Comment (USERSATOSHI) [inline src/functions/database/userLeaderBoard.js:38]: shouldnt this use Util.getMember instead of Util.getUser

**Review by USERSATOSHI — CHANGES_REQUESTED** [review]

**Review by USERSATOSHI — APPROVED** [review]
LGTM

**Review by Leref — APPROVED** [review]
Looks good to me.

**Pull request merged**